apps/admin/backend/test/app.ts

```diff
@@ -163,10 +162,17 @@ export async function configureMachine(
 }
 
 // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
-export function buildTestEnvironment() {
+export function buildTestEnvironment(workspaceRoot?: string) {
   const logger = fakeLogger();
   const auth = buildMockDippedSmartCardAuth();
-  const workspace = createWorkspace(tmp.dirSync().name);
+  const resolvedWorkspaceRoot =
+    workspaceRoot ??
```

Comment:
Should this be `||` instead of `??`? I think this is allowing the workspace root to be set to an empty string, but not sure if that's valid value

Reply:
That's a good point! I didn't realize those two operators had different behavior like that.

Reply:
9e2bbcefb96857da244e9ef3e86d1b1c786c3649